File: api/py/src/authentication_example.py
import json
import os
from base58 import b58decode
from requests import Request, Session

from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey

from authentication_example_signer import Signer


base_url = "https://testnet-api-evm.orderly.org"
orderly_account_id = "<orderly-account-id>"

key = b58decode(os.environ.get("ORDERLY_SECRET"))
orderly_key = Ed25519PrivateKey.from_private_bytes(key)

session = Session()
signer = Signer(orderly_account_id, orderly_key)

req = signer.sign_request(
    Request(
        "POST",
        "%s/v1/order" % base_url,
        json={
            "symbol": "PERP_ETH_USDC",
            "order_type": "MARKET",
            "order_quantity": 0.01,
            "side": "BUY",
        },
    )
)
res = session.send(req)
response = json.loads(res.text)

File: api/ts/src/authenticationExample.ts
import bs58 from 'bs58';
import { config } from 'dotenv';
import { webcrypto } from 'node:crypto';

import { signAndSendRequest } from './signer';

// this is only necessary in Node.js to make `@noble/ed25519` dependency work
// eslint-disable-next-line @typescript-eslint/no-explicit-any
if (!globalThis.crypto) globalThis.crypto = webcrypto as any;

config();

async function main() {
  const baseUrl = 'https://testnet-api-evm.orderly.org';
  const orderlyAccountId = '<orderly-account-id>';

  const orderlyKey = bs58.decode(process.env.ORDERLY_SECRET!);

  const res = await signAndSendRequest(orderlyAccountId, orderlyKey, `${baseUrl}/v1/order`, {
    method: 'POST',
    body: JSON.stringify({
      symbol: 'PERP_ETH_USDC',
      order_type: 'MARKET',
      order_quantity: 0.01,
      side: 'BUY'
    })
  });
  const response = await res.json();
  console.log(response);
}

main();
